akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala

```diff
@@ -46,33 +55,46 @@ final case class ClusterRouterGroupSettings(
   totalInstances:    Int,
   routeesPaths:      immutable.Seq[String],
   allowLocalRoutees: Boolean,
-  useRole:           Option[String]) extends ClusterRouterSettingsBase {
+  useRoles:          Set[String]) extends ClusterRouterSettingsBase {
```

Comment:
Adding 

```scala
    @deprecated("Use constructor with useRoles instead", since = "2.5.4")
    def copy(totalInstances: Int = totalInstances, routeesPaths: immutable.Seq[String] = routeesPaths, allowLocalRoutees: Boolean = allowLocalRoutees, useRole: Option[String] = useRole): ClusterRouterGroupSettings =
        ClusterRouterGroupSettings(totalInstances, routeesPaths, allowLocalRoutees, useRole)
```

prevents the use of the default copy generated for the case class. That means that 
```scala
    def withUseRoles(useRoles: Set[String]): ClusterRouterGroupSettings =
        copy(useRoles = useRoles)
``` 
does not compile. Is that what you have in mind ?